builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -124,6 +129,12 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"memory_reservation": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
```

Comment:
It can change on existing VMs with VMware vSphere. But it needs to be set ForceNew option because there is no support to update resources now.